Friend/firmware/firmware_v1.0/src/transport.c

```diff
@@ -67,7 +70,119 @@ static struct bt_gatt_attr dfu_service_attr[] = {
+static struct bt_uuid_128 accel_uuid_x = BT_UUID_INIT_128(BT_UUID_128_ENCODE(0x32403791,0x0000,0x1000,0x7450,0xBF445E5829A2));
+
+static void accel_ccc_config_changed_handler(const struct bt_gatt_attr *attr, uint16_t value);
+static ssize_t accel_data_read_characteristic(struct bt_conn *conn, const struct bt_gatt_attr *attr, void *buf, uint16_t len, uint16_t offset);
+
+static struct bt_gatt_attr accel_service_attr[] = {
+    BT_GATT_PRIMARY_SERVICE(&accel_uuid),//primary description
+    BT_GATT_CHARACTERISTIC(&accel_uuid_x.uuid, BT_GATT_CHRC_READ | BT_GATT_CHRC_NOTIFY, BT_GATT_PERM_READ, accel_data_read_characteristic, NULL, NULL),//data type
+    BT_GATT_CCC(accel_ccc_config_changed_handler, BT_GATT_PERM_READ | BT_GATT_PERM_WRITE),//scheduler
+};
+static struct bt_gatt_service accel_service = BT_GATT_SERVICE(accel_service_attr);
+
+static ssize_t accel_data_read_characteristic(struct bt_conn *conn, const struct bt_gatt_attr *attr, void *buf, uint16_t len, uint16_t offset)
+{
+    LOG_INF("Acceleration data read characteristic");
+    int axis_mode = 6; //3 for accel, 6 for (also) gyro
+    return bt_gatt_attr_read(conn, attr, buf, len, offset, &axis_mode, sizeof(axis_mode));
+}
+
+
+#define ACCEL_REFRESH_INTERVAL 1000 // 0.5 seconds
+
+void broadcast_accel(struct k_work *work_item);
+K_WORK_DELAYABLE_DEFINE(accel_work, broadcast_accel);
+
+void broadcast_accel(struct k_work *work_item) {
+
+    sensor_sample_fetch_chan(lsm6dsl_dev, SENSOR_CHAN_ACCEL_XYZ);
+	sensor_channel_get(lsm6dsl_dev, SENSOR_CHAN_ACCEL_X, &mega_sensor.a_x);
+	sensor_channel_get(lsm6dsl_dev, SENSOR_CHAN_ACCEL_Y, &mega_sensor.a_y);
+	sensor_channel_get(lsm6dsl_dev, SENSOR_CHAN_ACCEL_Z, &mega_sensor.a_z);
+
+    sensor_sample_fetch_chan(lsm6dsl_dev, SENSOR_CHAN_GYRO_XYZ);
+	sensor_channel_get(lsm6dsl_dev, SENSOR_CHAN_GYRO_X, &mega_sensor.g_x);
+	sensor_channel_get(lsm6dsl_dev, SENSOR_CHAN_GYRO_Y, &mega_sensor.g_y);
+	sensor_channel_get(lsm6dsl_dev, SENSOR_CHAN_GYRO_Z, &mega_sensor.g_z);
+
+   //only time mega sensor is changed is through here (hopefully),  so no chance of race condition
+    int err = bt_gatt_notify(current_connection, &accel_service.attrs[1], &mega_sensor, sizeof(mega_sensor));
+    if (err) {
+       LOG_INF("Error updating accelerometer data\n");
+    }
+    k_work_reschedule(&accel_work, K_MSEC(ACCEL_REFRESH_INTERVAL));
+}
+
+
+//use d4,d5
+static void accel_ccc_config_changed_handler(const struct bt_gatt_attr *attr, uint16_t value) {
+        if (value == BT_GATT_CCC_NOTIFY)
+    {
+        LOG_INF("Client subscribed for notifications");
+    }
+    else if (value == 0)
+    {
+        LOG_INF("Client unsubscribed from notifications");
+    }
+    else
+    {
+        LOG_INF("Invalid CCC value: %u", value);
+    }
+}
+
+int accel_start() {
+    struct sensor_value odr_attr;
+    lsm6dsl_dev = DEVICE_DT_GET_ONE(st_lsm6dsl);
+    k_msleep(50);
+    if (lsm6dsl_dev == NULL) {
+        LOG_INF("Could not get LSM6DSL device\n");
```

Comment:
Use `LOG_ERR` also remove all the `\n` from all the logs its not needed and polutes the log